src/movies.js

```diff
@@ -1,16 +1,99 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+const getAllDirectors = moviesArr => {
+    console.log('here')
+    let getDirectors = moviesArr.map(el => el.director);
+    return getDirectors;
+   
+};
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+let directors = getAllDirectors()
+
+const filteredDirectors = directors.filter((director, index) => directors.indexOf(director) === index);
+
+console.log(filteredDirectors)
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+const howManyMovies = moviesArr => {
+    let filteredDrama = moviesArr.filter(function (movie) { return movie.genre.includes('Drama') && movie.director === 'Steven Spielberg' })
+    return filteredDrama.length;
+};
+howManyMovies();
+  
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+const ratesAverage = arr => {
+    let theSum = movies.reduce((accum, currValue) => accum + currValue.rate, 0);
+  
+    let theAvg = theSum / movies.length;
+    
+    return theAvg.toFixed(2)
+  
+  };
+  ratesAverage()
+
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
+const avgDramaRate = moviesArr => {
```

Comment:
You need to use the same function names as they are described in tests. So here the correct name would be `dramaMoviesRate()`. Also you have passed the parameter `moviesArr` which you don't use in the function. Instead, you are using `movies`. These are small issues but it is important to keep an eye on everything.